pkg/metadata/host/host_tags.go

```diff
@@ -106,39 +113,37 @@ func GetHostTags(ctx context.Context, cached bool) *Tags {
 		return nil, nil
 	}
 
-	providers := make(map[string]*providerDef)
-
-	if config.Datadog.GetBool("collect_ec2_tags") {
-		// WARNING: if this config is enabled on a non-ec2 host, then its
-		// retries may time out, causing a 3s delay
-		providers["ec2"] = &providerDef{10, ec2.GetTags, false}
-	}
-
+	providers := getProvidersDefinitionsFunc()
+	// GCE is handled separately as it requires inlined var/function
 	if config.Datadog.GetBool("collect_gce_tags") {
 		providers["gce"] = &providerDef{1, getGCE, false}
 	}
 
```

Comment:
Why did this not move to getProvidersDefinitionsFunc with the others?

Reply:
There's a comment dedicated to explain that: https://github.com/DataDog/datadog-agent/pull/13741/files#diff-89147ccd687fc533906e97585d37f4d547e67ca70a9ee2d5440261f5ae7ea34dR117

Basically we do have a dedicated field for GCE (reason unknown to me), so you need to have an inline function to make it work.
The other 2 options are a global var or `if provider == gce` in the loop.

Reply:
I moved to the `if provider == gce` in the loop as it allows to re-use cache.